framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -167,12 +220,26 @@ private static boolean isMemberSelect(String s, FlowExpressionContext context) {
      * @param s expression string
      * @return pair of object and field
      */
-    private static Pair<String, String> parseMemberSelect(String s) {
-        Pair<Pair<String, String>, String> method = parseMethodCall(s);
-        if (method != null && method.second.startsWith(".")) {
-            return Pair.of(
-                    method.first.first + "(" + method.first.second + ")",
-                    method.second.substring(1));
+    private static Pair<String, String> parseMemberSelect(String s, FlowExpressionContext context) {
+
+        // replace occurence of formal parameters, such as "#2", with corresponding arguments
+        String sCopy = s;
+        int k = 0;
+        ArrayList<String> argumentList = new ArrayList<String>();
+        if (context.arguments != null) {
+            for (Receiver argument : context.arguments) {
+                String replaceParam = "#" + Integer.toString(k + 1);
+                sCopy = sCopy.replace(replaceParam, argument.toString());
+                k++;
+                argumentList.add(argument.toString());
+            }
+        }
+
+        // parses field accesses and method calls which contain a method call
```

Comment:
I don't understand this comment, but that's probably because I don't understand the new `parseMethodCall` routine.